sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/udf/GetFeedRangesForBuckets.scala

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.cosmos.spark.udf
+
+import com.azure.cosmos.implementation.SparkBridgeImplementationInternal
+import com.azure.cosmos.implementation.feedranges.FeedRangeEpkImpl
+import com.azure.cosmos.models.FeedRange
+import com.azure.cosmos.spark.CosmosPredicates.requireNotNullOrEmpty
+import org.apache.spark.sql.api.java.UDF2
+
+@SerialVersionUID(1L)
+class GetFeedRangesForBuckets extends UDF2[String, Int, Array[String]] {
```

Comment:
Name proposal: Maybe GetFeedRangesForContainer and bucketCount --> targetedCount and make this Option[Int] (in None just return FeedRanges)

Reply:
I am not sure about this name because I am not getting the feed ranges using the container. What should I return if targetedCount is None? Should I change it to take in a container as well, and that way I can return the feed ranges corresponding to the physical partitions when targeted count is None

Reply:
That was my idea - but i missed that you do no pass in the config dictionary (but the pkdefinition). I would probably change this to instead of pkdefinition add config dictionary paramter. PKDefinition is cached - so, you can get it easily from config - and that way zou can get the container if target count is null. Is that reasonable?